cli/pcluster/config/mapping.py

```diff
@@ -0,0 +1,530 @@
+                "default": 100,
+                "validator": raid_volume_iops_validator,
+            }),
+            ("encrypted", {
+                "type": BoolParam,
+                "default": False,
+            }),
+            ("ebs_kms_key_id", {}),
+        ]
+    )
+}
+
+
+FSX = {
+    "type": Section,
+    "key": "fsx",
+    "label": "default",
+    "validator": fsx_validator,
+    "cfn": "FSXOptions",  # All the parameters in the section are converted into a single CFN parameter
+    "params": OrderedDict(  # Use OrderedDict because the parameters must respect the order in the CFN parameter
+        [
+            ("shared_dir", {}),
+            ("fsx_fs_id", {
+                "allowed_values": r"^fs-[0-9a-z]{17}|NONE$",
+                "validator": fsx_id_validator,
+            }),
+            ("storage_capacity", {
+                "type": IntParam,
+                "validator": fsx_storage_capacity_validator
+            }),
+            ("fsx_kms_key_id", {}),
+            ("imported_file_chunk_size", {
+                "type": IntParam,
+                "validator": fsx_imported_file_chunk_size_validator
+            }),
+            ("export_path", {}),  # TODO add regex
+            ("import_path", {}),  # TODO add regex
+            ("weekly_maintenance_start_time", {}),  # TODO add regex
+        ]
+    )
+}
+
+CLUSTER = {
+    "type": ClusterSection,
+    "key": "cluster",
+    "label": "default",
+    "validator": cluster_validator,
+    "params": {
+        # Basic configuration
+        "key_name": {
+            "cfn": "KeyName",
+            "validator": ec2_key_pair_validator,
+        },
+        "template_url": {
+            # TODO add regex
+            "validator": url_validator,
+        },
+        "base_os": {
+            "default": "alinux",
+            "cfn": "BaseOS",
+            "allowed_values": ["alinux", "ubuntu1404", "ubuntu1604", "centos6", "centos7"],
+        },
+        "scheduler": {
+            "default": "sge",
+            "cfn": "Scheduler",
+            "allowed_values": ["awsbatch", "sge", "slurm", "torque"],
+            "validator": scheduler_validator,
+        },
+        "shared_dir": {
+            "type": SharedDirParam,
+            "cfn": "SharedDir",
+            "default": "/shared",
+        },
+        # Cluster configuration
+        "placement_group": {
+            "cfn": "PlacementGroup",
+            "validator": ec2_placement_group_validator,
+        },
+        "placement": {
+            "default": "compute",
+            "cfn": "Placement",
+            "allowed_values": ["cluster", "compute"],
+        },
+        # Master
+        "master_instance_type": {
+            "default": "t2.micro",  # TODO add regex or validator
+            "cfn": "MasterInstanceType",
+        },
+        "master_root_volume_size": {
+            "type": IntParam,
+            "default": 20,
+            "allowed_values": r"^([0-9]+[0-9]{2}|[2-9][0-9])$",  # >= 20
```

Comment:
optional: I can see we have many regexes to validate number ranges. We could have defined a class named Range and pass Range(min,max) to the allowed_values for the validation. Also this can be extended to IntRange, FloatRange

Reply:
We'll do in future ;)